dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/utils/TaskFilesTransferUtils.java

```diff
@@ -183,7 +210,11 @@ public static void downloadUpstreamFiles(TaskExecutionContext taskExecutionConte
                 logger.info("{} --- Remote:{} to Local:{}", property, resourceWholePath, downloadPath);
                 storageOperate.download(taskExecutionContext.getTenantCode(), resourceWholePath, downloadPath, false,
                         true);
-            } catch (IOException ex) {
+
+                if (isScpCommandTemplate) {
+                    isPack = scpFetchFile(downloadPath, targetPath);
+                }
```

Comment:
If upstreaming task uses cache mechanism, scp fetch will fail, however, transferred files have been uploaded to resource center. Try to remove template suffix of resource path and download them.